src/pages/writing-nft/index.vue

```diff
@@ -213,17 +216,49 @@ export default {
       this.$axios.$get(getTopNFTClasses()),
       this.$axios.$get(getLatestNFTClasses()),
     ]);
-    [this.trendingClassIds, this.latestClassIds] = [trendingRes, latestRes].map(
-      res =>
-        (res.classes || [])
-          .filter(
-            c => c.metadata?.nft_meta_collection_id === 'likerland_writing_nft'
-          )
-          .map(c => c.id)
-          .slice(0, 10)
+    const [trendingClasses, latestClasses] = [trendingRes, latestRes].map(res =>
+      (res.classes || []).filter(
+        c => c.metadata?.nft_meta_collection_id === 'likerland_writing_nft'
+      )
     );
+    this.latestClassIds = latestClasses
+      .slice(0, NFT_CLASS_DISPLAY_COUNT)
+      .map(c => c.id);
+    this.trendingClassIds = await this.filterNFTClassesByOwner(trendingClasses);
   },
   methods: {
+    async getOwnerFromUri(uri) {
+      try {
+        const data = await this.$axios.$get(uri);
+        return data.iscn_owner;
+      } catch (err) {
+        console.error(`Failed to fetch uri ${uri}`);
+        return null;
+      }
+    },
+    async filterNFTClassesByOwner(nftClasses) {
+      const ownerList = await Promise.all(
+        nftClasses
+          .slice(0, NFT_CLASS_DISPLAY_COUNT * 3)
+          .map(c => this.getOwnerFromUri(c.uri))
+      );
+      const filteredNFTClasses = [];
+      const ownerToNFTClassCountMap = {};
+      for (let i = 0; i < ownerList.length; i += 1) {
+        const owner = ownerList[i];
+        // eslint-disable-next-line no-continue
+        if (!owner) continue;
```

Comment:
This case should not exists, but for NFT without owner we should just show it without filtering? Since we just want to handle the case of many entries from same owner